src/tracy.c

```diff
@@ -152,3 +161,15 @@ static char const * get_error_string(int err) {
   return strerror_r(err, errbuf, MAX_ERR_STR_BUF_SIZE);
 #endif
 }
+
+void trc_private_print(char const * fmt, ...) {
+  va_list args;
+  va_start(args, fmt);
+  if (print_callback) {
```

Comment:
No need for special cases. Write a default print callback that uses `vprintf` and set it to the print callback. Then you should just assert that the callback handler is not NULL.